### DISCUSSION TITLE
Use of weight from all_shortest_paths

### DISCUSSION THREAD

**Discussion by MaxDragonheart** [networkx/networkx#6128] [Q&A] [answered]
I've a doubt about the use of `all_shortest_paths`. I've a list of edges like this:
```
edge_list = [
    (from_nodeA, to_nodeB, {'id': NUMBER, 'name': TEXT, 'cost': NUMBER}),
    (from_nodeB, to_nodeC, {'id': NUMBER, 'name': TEXT, 'cost': NUMBER}),
    (from_nodeA, to_nodeC, {'id': NUMBER, 'name': TEXT, 'cost': NUMBER}),
....
....
....
]
```
With the code below I've created a graph  and I use it to find the shortest paths:
```
graph = ntx.MultiDiGraph()
graph.add_edges_from(edge_list)

all_shortest_paths(
        G=graph,
        source=from_node,
        target=to_node,
        weight='cost',
        method='dijkstra',
)
```
If I use `all_shortest_paths` with 'cost' as *weight*, the function find all shortest paths based on `cost` present into edges' attributes?  It's right?

**Comment by MridulS** [comment, marked as answer]
> If I use all_shortest_paths with 'cost' as weight, the function find all shortest paths based on cost present into edges' attributes? It's right?

Yes it will minimize the "length" of the path by using the `cost` attribute of the edge and return all the shortest paths.

**Reply by MaxDragonheart** [reply]
Thank you
